mesonbuild/environment.py

```diff
@@ -1285,7 +1263,26 @@ def libdir_layout_is_win(self):
         return self.is_windows() \
             or self.is_cygwin()
 
-class MachineInfos(PerMachine):
+    # TODO make this compare two `MachineInfo`s purely. How important is the
+    # `detect_cpu_family({})` distinction? It is the one impediment to that.
+    def can_run(self):
+        """Whether we can run binaries for this machine on the current machine.
+
+        Can almost always run 32-bit binaries on 64-bit natively if the host
+        and build systems are the same. We don't pass any compilers to
+        detect_cpu_family() here because we always want to know the OS
+        architecture, not what the compiler environment tells us.
+        """
+        if self.system != detect_system():
+            return False
+        true_build_cpu_family = detect_cpu_family({})
+        return \
+            (self.cpu_family == true_build_cpu_family) or \
+            ((true_build_cpu_family == 'x86_64') and (self.cpu_family == 'x86'))
```

Comment:
since you're already using parens here, please use more parens instead of \
```python
return ((self.cpu_family == true_build_cpu_family) or
        ((true_build_cpu_family == 'x86_64') and (self.cpu_family == 'x86')))
```

Reply:
I had done this, but flake8 was complaining about other things. Sorry I wish I had written it the exact message but it seemed like the non-`\` was actually going to be quite a lot of hoops, so I changed it bak.